README.md

```diff
@@ -1,69 +1,77 @@
-# Dependency Injection Checker
+# Guice Checker
 
-A common problem when programming is TODO.
-This results in a run-time exception.
-
-The Dependency Injection Checker guarantees, at compile time, that your code will
-not suffer that run-time exception.
+A common challenge when building systems that use [Guice](https://github.com/google/guice) is dealing with dependencies and ensuring they are properly defined. Omitting or misconfiguring dependencies can lead to runtime errors, impacting the reliability and stability of code.
 
+The Guice Checker is a static analysis tool designed to address this issue by providing compile-time guarantees that your Guice dependency mappings are accurately defined.
 
 ## How to run the checker
 
-First, publish the checker to your local Maven repository by running
-`./gradlew publishToMavenLocal` in this repository.
+First, build the checker by running `./gradlew clean build` and then publish the checker to your local Maven repository by running `./gradlew publishToMavenLocal` in this repository.
 
 Then, if you use Gradle, add the following to the `build.gradle` file in
-the project you wish to type-check (using Maven is similar):
+the project you wish to type-check:
 
 ```
-repositories {
-    mavenLocal()
-    mavenCentral()
+plugins {
+  id 'org.checkerframework' version '0.6.28'
+  ...
+}
+
+...
+
+apply plugin: 'org.checkerframework'
+checkerFramework {
+  checkers = [
+    'org.checkerframework.checker.dependencyinjection.DependencyInjectionChecker',
+  ]
 }
+
+...
+
 dependencies {
-    annotationProcessor 'org.checkerframework:dependencyinjection-checker:0.1-SNAPSHOT'
+    compileOnly 'org.checkerframework:dependencyinjection-checker-qual:0.1-SNAPSHOT'
+    testCompileOnly 'org.checkerframework:dependencyinjection-checker-qual:0.1-SNAPSHOT'
+    checkerFramework 'org.checkerframework:dependencyinjection-checker:0.1-SNAPSHOT'
+    ...
 }
 ```
 
-Now, when you build your project, the Dependency Injection Checker will also run,
-informing you of any potential errors related to TODO.
+Now, when you build your project, the Guice Checker will also run, informing you of any potential errors related to improperly defined or configured bindings.
 
 
 ## How to specify your code
```

Comment:
The actual text here is more like a description of "How the tool works" than "How to specify your code", so I suggest changing the title of this section.